src/auth/auth.ts

```diff
@@ -1,7 +1,16 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
 import { AxiosInstance } from 'axios';
 
 export default interface Auth {
     // Hook into axios and intercept requests to add Auth tokens
     // https://masteringjs.io/tutorials/axios/interceptors
-    attachInterceptor(axiosInstance: AxiosInstance): void;
+    attachInterceptor(axiosInstance: AxiosInstance, interceptorConfig: InterceptorConfig): void;
 }
+
+export type InterceptorConfig = {
```

Comment:
I added this type, because for the `Auth` interface, we might need other config values in the future. For example, we might need to pass an `api key` to an `Auth` implementation in the future.